server/workflowarchive/archived_workflow_server.go

```diff
@@ -298,13 +303,31 @@ func (w *archivedWorkflowServer) RetryArchivedWorkflow(ctx context.Context, req
 				return nil, sutils.ToStatusError(err, codes.Internal)
 			}
 		}
+		large, _ := packer.IsLargeWorkflow(wf)
+		if large {
+			log.WithFields(log.Fields{"Dehydrate workflow uid=": wf.UID}).Info("RetryArchivedWorkflow")
+			err = w.hydrator.Dehydrate(wf)
+			if err != nil {
+				return nil, sutils.ToStatusError(err, codes.Internal)
+			}
+		}
 
 		wf.ObjectMeta.ResourceVersion = ""
 		wf.ObjectMeta.UID = ""
 		result, err := wfClient.ArgoprojV1alpha1().Workflows(req.Namespace).Create(ctx, wf, metav1.CreateOptions{})
 		if err != nil {
 			return nil, sutils.ToStatusError(err, codes.Internal)
 		}
+		if large {
+			offloadedNodes, err := w.offloadNodeStatusRepo.Get(string(oriUid), wf.GetOffloadNodeStatusVersion())
```

Comment:
what if wf.GetOffloadNodeStatusVersion() is "" ?

Reply:
If we want to retry a archived workflow, it will create a new workflow, it is different to retry a not-archived workflow. So when retry large archived workflow, we need to `Dehydrate` the wf ( offload the node status) first, the key of these node status is the `uid` of workflow, and then we need to create a new workflow according the Dehydrated workflow. And as for the controller, it will read the offload-node status by the `uid` of the new created workflow, so we copy the node status info of the archived workflow and save it with new `uid` of the new  created workflow. we could only get the `uid` after creating the new workflow, and the controller may process the new created workflow before we update node status with new `uid`, so we catch the db error and make it retry.

Reply:
I known we should get offloadedNodes. But if version is "", we got err:
```
time="2024-03-12T14:01:23.864Z" level=info msg="Workflow to be dehydrated" Workflow Size=3361331
time="2024-03-12T14:01:24.370Z" level=debug msg="Getting offloaded nodes" uid=06ab13ed-7e5a-4a0c-aeae-40c19f196bc5 version=
time="2024-03-12T14:01:24.371Z" level=error msg="finished unary call with code Internal" error="rpc error: code = Internal desc = upper: no more rows in this result set" grpc.code=Internal grpc.method=RetryArchivedWorkflow grpc.service=workflowarchive.ArchivedWorkflowService grpc.start_time="2024-03-12T14:01:23Z" grpc.time_ms=1242.014 span.kind=server system=grpc
```
you can test it argo server UI.

Reply:
Maybe we don't need to check "large", becaust Dehydrate->CompressWorkflowIfNeeded>IsLarge have.

Reply:
> Maybe we don't need to check "large", becaust Dehydrate->CompressWorkflowIfNeeded>IsLarge have.

yeah, I have removed this check 'large'.